### PR TITLE
docs: formalized '1 approval' and 'team assigned' labels

### LINKED ISSUE #9423: Docs: Decide how we want to use '1 approval' and 'team assigned' labels
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Since we added new committer team members earlier this year, we've been making more liberal use of two labels:

* `1 approval`: which can mean either or both of:
  * Ready to go whenever someone wants to merge
  * Waiting for a second team approval
* `team assigned`: an alternative to `accepting prs` that roughly means the same thing, but with the stipulation that someone on the team should be the one to send a PR

`1 approval` is only lightly documented. `team assigned` is new and hasn't been documented at all. We've had some private chats about how we'd like to clarify at least the processes around `1 approval`.

### Affected URL(s)

https://typescript-eslint.io/maintenance/pull-requests

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9423
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Summarizing scattered conversations I've had over the last few months. This PR is very much a proposal, I'm expecting some discussions in it. 😄 

💖

### CODE REVIEW

File: docs/maintenance/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [inline docs/maintenance/Pull_Requests.mdx:120]: I couldn't figure out how to phrase the idea succinctly. I.e. I want to make sure that reviewers who only have 1-2 days a week to review get a chance to see things.

This has been an issue for me in the past in other projects (not this one), where things get shoved through by 2 people in a small number of business days and I never got a chance to give input. 😞

Comment (kirkwaiblinger) [reply docs/maintenance/Pull_Requests.mdx:120]: Maybe like this?
```suggestion
We try to wait a week on PRs to give reviewers who aren't active every day a chance to get to it.
```

**Review by kirkwaiblinger — APPROVED** [review]
i like

**Review by kirkwaiblinger — APPROVED** [review]

File: docs/maintenance/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [inline docs/maintenance/Pull_Requests.mdx:116]: Oh, I wonder if we should add a contingency for urgent things, like security reports or broken `main` builds...

**Review by bradzacher — APPROVED** [review]

File: docs/maintenance/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [reply docs/maintenance/Pull_Requests.mdx:118]: 3abac143f

File: docs/maintenance/Pull_Requests.mdx
Comment (Josh-Cena) [inline docs/maintenance/Pull_Requests.mdx:125]: This seems redundant? What's the purpose of this from the bullet above.

**Review by Josh-Cena — APPROVED** [review]

**Review by bradzacher — APPROVED** [review]

File: docs/maintenance/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [reply docs/maintenance/Pull_Requests.mdx:125]: My thought was that the instructions technically only apply if you are reviewing... but, yeah, I think anybody on the team should be able to understand this is implied. 👍

**Review by Josh-Cena — DISMISSED** [review]

File: docs/maintenance/Pull_Requests.mdx
Comment (kirkwaiblinger) [inline docs/maintenance/Pull_Requests.mdx:120]: I guess we could also posit something along the lines of "no need to reset the 3 day timer for minor fixups from code reviews of otherwise approved code" or something, if that makes sense

File: docs/maintenance/Pull_Requests.mdx
Comment (JoshuaKGoldberg) [inline docs/maintenance/Pull_Requests.mdx:124]: Made this a bit more clear/precise: it's a week in review, not a week since approval. IMO no need to wait a week after approval for something that's already been open for a bit.

Comment (kirkwaiblinger) [reply docs/maintenance/Pull_Requests.mdx:124]: Oh, good call on clarifying that. That's a better approach than what my understanding was! 👍

**Review by kirkwaiblinger — APPROVED** [review]

**Pull request merged**
